### ISSUE TITLE
NPE in `getLast()`

### ISSUE THREAD

**Issue by avernet** [orbeon/orbeon-forms#6016]
To reproduce this, paste this [form](https://github.com/orbeon/orbeon-forms/files/12877409/form.txt) into Form Builder, test it, and click the "Insert data at end" button.

The call to `last()` in XPath ends up in `getLast()` of `XPathContextMinor.java`, from Saxon 9.1, which calls `currentIterator.getAnother()`. From `SequenceIterator.java`, `getAnother()` is expected to return "another SequenceIterator that iterates over the same items as the original, but which is repositioned at the start of the sequence", but in our implementation thereof in `Implicits.scala` we have `def getAnother : SequenceIterator = null`, because we couldn't "rewind" the iterator. I think that unfortunately, if we need to be able to "rewind", we need to preemptively convert the `Iterator` to a `Seq`. This solves the problem, simplifies our code, but will be less efficient if `getAnother()` ends up not being called.

[+1 from customer](https://groups.google.com/g/orbeon/c/FteRlyPw80k)

**Comment by ebruchez** [comment]
Noting that `getAnother()` is gone starting Saxon 9.6. So this is a legacy issue, hopefully. See also #5107.

**Comment by ebruchez** [comment]
I fixed this by checking and removing uses of the implicit conversion from Scala `Iterator`s to a Saxon `SequenceIterator`.

- Some uses were or could be made explicit, and I moved that to `SaxonUtils` for `deepCompare()` only, which doesn't hit the case where `getAnother` is needed.
- Other uses now convert the `Iterator`s to `Seq` first.  Those shouldn't return very large sequences.
- I eliminated some use completely by returning a `String` or a `Boolean`.

On the JS side, which uses the Saxon 11 XPath implementation, I did some cleanup of uses and implicits.
